src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Data;
+using System.Data.SqlClient;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    public class SqlServerCache : IDistributedCache
+    {
+        internal const int ExpirationTimeMultiplier = 2;
+
+        // Table column names
+        private const string CacheItemId = "Id";
+        private const string CacheItemValue = "Value";
+        private const string ExpiresAtTimeUTC = "ExpiresAtTimeUTC";
+        private const string SlidingExpirationInTicks = "SlidingExpirationInTicks";
+
+        private readonly SqlServerCacheOptions _options;
+        private readonly ILogger _logger;
+        private readonly ISystemClock _systemClock;
+        private readonly SqlQueries _sqlQueries;
+        private DateTimeOffset _lastExpirationScan;
+
+        //TODO: ADD LOGGING
+        public SqlServerCache(IOptions<SqlServerCacheOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options.Options;
+            _systemClock = _options.SystemClock;
+            _sqlQueries = new SqlQueries(_options.TableName);
+            _logger = loggerFactory.CreateLogger<SqlServerCache>();
+        }
+
+        public void Connect()
+        {
+            //TODO:
```

Comment:
How about performing a no-op operation to verify the connection string and server availability?